src/components/pages/import.js

```diff
@@ -0,0 +1,149 @@
+import by from 'thenby';
+import Page from '../page.js';
+import PropTypes from 'prop-types';
+import urls from '../../utils/urls.js';
+import React, { useState } from 'react';
+import Logger from '../../common/logger.js';
+import { useTranslation } from 'react-i18next';
+import { Tooltip, Button, Table, Select } from 'antd';
+import { inject, useService } from '../container-context.js';
+import { DOCUMENT_IMPORT_TYPE } from '../../common/constants.js';
+import ImportApiClient from '../../services/import-api-client.js';
+import { useDateFormat, useLanguage } from '../language-context.js';
+import LanguageNameProvider from '../../data/language-name-provider.js';
+import CountryFlagAndName from '../localization/country-flag-and-name.js';
+import { CloudDownloadOutlined, CloudSyncOutlined } from '@ant-design/icons';
+
+const logger = new Logger(import.meta.url);
+
+function Import({ initialState, importApiClient }) {
+  const { t } = useTranslation('import');
+
+  const { language } = useLanguage();
+  const { formatDate } = useDateFormat();
+  const languageNameProvider = useService(LanguageNameProvider);
+
+  const sourceMenuItems = initialState.importSources;
+  const [selectedSource, setSelectedSource] = useState(null);
+  const [selectedDocumentKeys, setSelectedDocumentKeys] = useState([]);
+  const [importableDocuments, setImportableDocuments] = useState([]);
+  const [isFetchingImportableDocuments, setIsFetchingImportableDocuments] = useState(false);
+
+  const handleImportClick = () => {
+    const tasks = selectedDocumentKeys
+      .map(key => importableDocuments.find(doc => doc.key === key))
+      .map(doc => ({
+        key: doc.key,
+        importedRevision: doc.importedRevision,
+        importableRevision: doc.importableRevision
+      }));
+    logger.info(`Dummy import sets tasks: ${JSON.stringify(tasks)}`);
+  };
+
+  const handleSourceMenuChange = async value => {
+    setIsFetchingImportableDocuments(true);
+    const newSelectedSource = sourceMenuItems.find(source => source.name === value);
+    setSelectedSource(newSelectedSource);
+    setImportableDocuments([]);
+    setSelectedDocumentKeys([]);
+
+    const { documents } = await importApiClient.getImports(newSelectedSource.name);
+    setImportableDocuments(documents);
+    setIsFetchingImportableDocuments(false);
```

Comment:
Please try catch finally this, else we risk the table always loading. Maybe we can also use the handleError() component.

Reply:
I know, I was thinking if this fails we don't want to just swallow the exception, cause the problem is then bigger than just the spinner running.
Wanted to check handleError() but I can't find it, btw.

Reply:
Sorry, it's called handleApiError.

Reply:
Handled, thanks!